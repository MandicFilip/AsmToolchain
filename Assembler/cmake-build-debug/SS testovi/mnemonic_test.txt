.global a, b

.text
    moval r0, a
    moveq r1, $a
    movne b, r2
    movgt $b, r3

    addal r0, r1
    subal r0, r1
    mulal r0, a
    dival r0, $a

    andal r1, r0[5]
    oral r1, r0[b]
    notal r0, *20
    testal $a, r0

    cmpal r0, &b

    shlal r1, &b
    shral r0, 20
    moval r1, 20

    pushal r0
    pushal a
    pushal $a
    popal a
    popal $a
    popal a

    callal a
    callal b

    jmpal a
    jmpal $a

    retal

    iretal
.end
