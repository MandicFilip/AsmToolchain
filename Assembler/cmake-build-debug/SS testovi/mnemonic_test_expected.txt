
---Symbol Table---
Symbol   Id   Global   Defined    Value   Section    Offset
UND      0    0        1          0       UND        0
.text    1    0        1          a       text       0
a        2    1        0          0       UND        0
b        3    1        0          0       UND        0

#.ret.text
0002 apsolute 0002
0006 relative 0002
000a apsolute 0003
000e relative 0003
0016 apsolute 0002
001a relative 0002
0022 apsolute 0003
002a relative 0002
002e apsolute 0003
0032 apsolute 0003
0040 apsolute 0002
0044 relative 0002
0048 apsolute 0002
004c relative 0002
0050 apsolute 0002
0054 apsolute 0002
0058 apsolute 0003
005c apsolute 0002
0060 relative 0002

#.text
10 f5 00 00
30 35 fe ff
0a 76 00 00
0b b6 fe ff
09 c1
09 c5
10 c9 00 00
10 cd fe ff
38 d5 05 00
38 d9 00 00
10 dd 14 00
08 e2 fe ff
00 d1 00 00
20 f9 00 00
00 fd 14 00
20 f5 14 00
08 e4
10 e4 00 00
10 e4 00 00
00 ea 00 00
00 ea 00 00
00 ea 00 00
10 ec 00 00
10 ec 00 00
f0 f5 00 00
f0 c1 fe ff
e0 e9
00 f0
